### PR TITLE
WIP: Fix textarea in/valid icon getting left behind when a user resizes the field

### PR DESCRIPTION
**Pull request by snowystinger**
cases:
normal-top label:
nothing: resize height/width, container width fit-content
quiet: no resize
custom width: only resize height
custom height: only resize width
custom both: no resize

normal-side label:
nothing: resize height/width, container fit-content
quiet: no resize
custom width: only resize height
custom height: only resize width
custom both: no resize

in form-top label:
nothing: resize height, follow form width
quiet: no resize

in form-side label:
nothing: resize height, follow form width
quiet: no resize

Closes https://jira.corp.adobe.com/browse/RSP-1792

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/form/stories/Form.stories.tsx [hunk 1]
```diff
@@ -88,6 +88,14 @@ storiesOf('Form', module)
     'isQuiet, labelPosition: side',
     () => render({isQuiet: true, labelPosition: 'side'})
   )
+  .add(
+    'isQuiet, labelPosition: side, invalid',
+    () => render({isQuiet: true, labelPosition: 'side', validationState: 'invalid'})
+  )
```
Comment (LFDanLu) [inline packages/@react-spectrum/form/stories/Form.stories.tsx:94]: I noticed that the quiet textarea has the resize drag handle here, is that intentional?

Comment (LFDanLu) [reply packages/@react-spectrum/form/stories/Form.stories.tsx:94]: IMO quiet textareas shouldn't be resizable via a drag handle

File: packages/@react-spectrum/textfield/src/TextFieldBase.tsx [hunk 2]
```diff
@@ -136,9 +147,12 @@ function TextFieldBase(props: TextFieldBaseProps, ref: Ref<TextFieldRef>) {
...
               'spectrum-Textfield-input',
               {
-                'spectrum-Textfield-inputIcon': icon
+                'spectrum-Textfield-inputIcon': icon,
+                'spectrum-Textfield--resizableX': isResizeableWidth,
+                'spectrum-Textfield--resizableY': isResizeableHeight
               },
-              inputClassName
+              inputClassName,
+              classNames(formFieldStyles, {'spectrum-Form-input': isInForm})
```
Comment (LFDanLu) [inline packages/@react-spectrum/textfield/src/TextFieldBase.tsx:155]: related to the above comments, maybe slap a `&& !isQuiet` here as well
```suggestion
              classNames(formFieldStyles, {'spectrum-Form-input': isInForm && !isQuiet})
```

File: packages/@react-spectrum/form/src/Form.tsx [hunk 3]
```diff
@@ -45,7 +45,8 @@ function Form(props: SpectrumFormProps, ref: DOMRef<HTMLFormElement>) {
   let ctx = {
     labelPosition,
     labelAlign,
-    necessityIndicator
+    necessityIndicator,
+    isInForm: true
```
Comment (devongovett) [inline packages/@react-spectrum/form/src/Form.tsx:49]: Can we do this in CSS somehow instead?

**Comment by snowystinger** [comment]
Another case noted, label end-aligned needs to line up with textarea's edge, at least to start

**Comment by snowystinger** [comment]
should be fine to use https://caniuse.com/#feat=mutationobserver

**Comment by snowystinger** [comment]
Hooray, this is no longer an issue https://github.com/adobe/react-spectrum/pull/1543

**Pull request closed without merging**